lib/features/diary_detail/bloc/diary_display_content_cubit.dart

```diff
@@ -0,0 +1,55 @@
+import 'dart:convert';
+
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:flutter_quill/flutter_quill.dart';
+import 'package:interactive_diary/features/home/data/diary_display_content.dart';
+import 'package:interactive_diary/service_locator/service_locator.dart';
+import 'package:nartus_storage/nartus_storage.dart';
+
+part 'diary_display_content_state.dart';
+
+class DiaryDisplayContentCubit extends Cubit<DiaryDisplayContentState> {
+
+  final StorageService storageService = ServiceLocator.instance.get<StorageService>();
+
+  DiaryDisplayContentCubit() : super(DiaryDisplayContentInitial());
+
+  void fetchDiaryDisplayContent(int dateTime, String countryCode, String postalCode) async {
+
+    Diary? diary = await storageService.getDiary(
+        dateTime: dateTime,
+        countryCode: countryCode,
+        postalCode: postalCode,
+        month: DateTime.now());
+
+    if (diary == null) {
+      emit(DiaryDisplayContentNotFound());
+    } else {
+      String plainText = '';
+      String imageUrl = '';
+
+      for (Content content in diary.contents) {
+        if (content is TextDiary) {
+          final textJson = jsonDecode(content.description);
+          Document document = Document.fromJson(textJson);
+          plainText = '${document.toPlainText()}\n';
+        } else if (content is ImageDiary) {
+          imageUrl = content.thumbnailUrl;
```

Comment:
I think we should remove this thumbnailUrl for images, since we already have limit on photo size. Let's just use the photo URL instead